cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -430,7 +433,23 @@ template <>
 hash_value_type CUDA_DEVICE_CALLABLE
 SparkMurmurHash3_32<numeric::decimal128>::operator()(numeric::decimal128 const& key) const
 {
-  return this->compute<__int128_t>(key.value());
+  __int128_t const val    = key.value();
+  __int128_t flipped      = 0;
```

Comment:
It's a very awkward transition because of dynamic data length from the data shortening. I couldn't get clean data flipping while also maintaining common code. Let me dredge up/fix what the kernel might look like if we used a `swap_endian` like function.

Reply:
If the expansion of the compute call is ok, the following code will work.

```
  for (; length >= 0 && data[length] == zero_value; --length)
    ;
  if (length == -1) return this->compute<uint8_t>(0);
  if (length != 15 && sign_bit ^ (data[length] & static_cast<int8_t>(0x80))) ++length;
  //data contains the valuable data from index 0 to length
  //block 1 comes from the 4 right most bytes

  cudf::size_type len = length + 1;
  int32_t const nblocks = len / 4;
  uint32_t h1           = m_seed;
  constexpr uint32_t c1 = 0xcc9e2d51;
  constexpr uint32_t c2 = 0x1b873593;
  
  //----------
  // body
  uint32_t const* const blocks = reinterpret_cast<uint32_t const*>(data+len);//out of bounds but should ever try to access 0 index
  for (int i = -1; i >= -nblocks; i--) {
    uint32_t k1 = swap_endian(blocks[i]);
    k1 *= c1;
    k1 = rotl32(k1, 15);
    k1 *= c2;
    h1 ^= k1;
    h1 = rotl32(h1, 13);
    h1 = h1 * 5 + 0xe6546b64;
  }
  //----------
  // byte by byte tail processing
  for (int i = len - (nblocks * 4) - 1; i >= 0; i--) {
    int32_t k1 = data[i];
    k1 *= c1;
    k1 = rotl32(k1, 15);
    k1 *= c2;
    h1 ^= k1;
    h1 = rotl32(h1, 13);
    h1 = h1 * 5 + 0xe6546b64;
  }
  //----------
  // finalization
  h1 ^= len;
  h1 = fmix32(h1);
  return h1;```

Reply:
Resolved per outside discussion, but with vastly expanded comments and direct reference to the conversion that it mimics.